src/routes/LandingPage.tsx

```diff
@@ -99,11 +104,26 @@ const InReviewFlag = styled(FlexRowAlignCenter)`
 
 export default function LandingPage(): ReactElement {
   const theme = useContext(AppThemeContext);
+  const navigate = useNavigate();
+
+  // Behavior
+
+  const onDatasetLoad = (collection: Collection, datasetKey: string, _newDataset: Dataset): void => {
+    // Unfortunately we can't pass the dataset directly through the `navigate` `state` API due to
+    // certain Dataset state (like HTMLImageElement objects) being non-serializable. This means that the
+    // dataset will be loaded twice, once here and once in the viewer.
+    // Dataset loading is relatively fast and the browser should cache most of the loaded data so it
+    // should hopefully not be a performance issue.
+    // TODO: Pass dataset directly here?
+    navigate(PageRoutes.VIEWER, { state: { collection: collection, datasetKey: datasetKey } as LocationState });
+  };
+
+  // Rendering
 
   // TODO: Should the load buttons be link elements or buttons?
   // Currently both the link and the button inside can be tab-selected.
   const renderDataset = (dataset: DatasetEntry, index: number): ReactElement => {
-    const viewerLink = "viewer" + paramsToUrlQueryString(dataset.loadParams);
+    const viewerLink = PageRoutes.VIEWER + paramsToUrlQueryString(dataset.loadParams);
```

Comment:
personal preference: 
```suggestion
    const viewerLink = `${PageRoutes.VIEWER}${paramsToUrlQueryString(dataset.loadParams}`;
```